src/logic/tokens/utils/humanReadableValue.ts

```diff
@@ -3,3 +3,17 @@ import { BigNumber } from 'bignumber.js'
 export const humanReadableValue = (value: number | string, decimals = 18): string => {
   return new BigNumber(value).times(`1e-${decimals}`).toFixed()
 }
+
+export const fromTokenUnit = (amount: string, decimals: string | number): string =>
+  new BigNumber(amount).times(`1e-${decimals}`).toFixed()
+
+export const toTokenUnit = (amount: string, decimals: string | number): string => {
+  const amountBN = new BigNumber(amount).times(`1e${decimals}`)
+  const [, amountDecimalPlaces] = amount.split('.')
+
+  if (amountDecimalPlaces?.length >= +decimals) {
+    return amountBN.toFixed(0, BigNumber.ROUND_DOWN)
```

Comment:
are you sure about `.toFixed(0`? it won't return any decimals

Reply:
@fernandomg do you know the purpose of that?

Reply:
oh, that's plain wrong! have a look: https://github.com/gnosis/safe-react/blob/feature/%23413-SpendingLimit-in-app/src/routes/safe/components/Settings/SpendingLimit/utils.ts#L15-L38

`adjustAmountToToken` has the correct code.

Just change that `0` with the `+decimals` value

```suggestion
    return amountBN.toFixed(+decimals, BigNumber.ROUND_DOWN)
```

That's why is so important the code review. :grin: